src/main/java/net/imagej/ops/threshold/apply/LocalThresholdIntegral.java

```diff
@@ -0,0 +1,186 @@
+ * POSSIBILITY OF SUCH DAMAGE.
+ * #L%
+ */
+
+package net.imagej.ops.threshold.apply;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.scijava.plugin.Parameter;
+
+import net.imagej.ops.Ops;
+import net.imagej.ops.Ops.Map;
+import net.imagej.ops.map.neighborhood.CenterAwareIntegralComputerOp;
+import net.imagej.ops.special.computer.AbstractUnaryComputerOp;
+import net.imagej.ops.special.computer.BinaryComputerOp;
+import net.imglib2.FinalInterval;
+import net.imglib2.IterableInterval;
+import net.imglib2.RandomAccessibleInterval;
+import net.imglib2.algorithm.neighborhood.RectangleShape;
+import net.imglib2.algorithm.neighborhood.RectangleShape.NeighborhoodsIterableInterval;
+import net.imglib2.outofbounds.OutOfBoundsBorderFactory;
+import net.imglib2.outofbounds.OutOfBoundsFactory;
+import net.imglib2.type.logic.BitType;
+import net.imglib2.type.numeric.RealType;
+import net.imglib2.view.Views;
+import net.imglib2.view.composite.Composite;
+
+/**
+ * Apply a local thresholding method to an image using integral images for speed
+ * up, optionally using a out of bounds strategy.
+ *
+ * @author Stefan Helfrich (University of Konstanz)
+ */
+public abstract class LocalThresholdIntegral<I extends RealType<I>> extends
+	AbstractUnaryComputerOp<RandomAccessibleInterval<I>, IterableInterval<BitType>>
+{
+
+	@Parameter
+	protected RectangleShape shape;
+
+	@Parameter(required = false)
+	private OutOfBoundsFactory<I, RandomAccessibleInterval<I>> outOfBoundsFactory =
+		new OutOfBoundsBorderFactory<>();
+
+	private CenterAwareIntegralComputerOp<I, BitType> filterOp;
+
+	private BinaryComputerOp<RandomAccessibleInterval<I>, NeighborhoodsIterableInterval<? extends Composite<RealType>>, IterableInterval<BitType>> map;
+
+	@Override
+	public void initialize() {
+		// Increase span of shape by 1 to return correct values together with
+		// the integralSum operation
+		shape = new RectangleShape(shape.getSpan() + 1, false);
+
+		filterOp = unaryComputer();
+	}
+
+	@SuppressWarnings({ "unchecked", "rawtypes" })
+	@Override
+	public void compute1(RandomAccessibleInterval<I> input,
+		IterableInterval<BitType> output)
+	{
+
+		List<RandomAccessibleInterval<RealType>> listOfIntegralImages =
+			new ArrayList<>();
+		for (int order : requiredIntegralImages()) {
+			RandomAccessibleInterval<RealType> requiredIntegralImg = getIntegralImage(
+				input, order);
+			listOfIntegralImages.add(requiredIntegralImg);
+		}
+
+		// Composite image of integral images of order 1 and 2
+		RandomAccessibleInterval<RealType> stacked = Views.stack(
+			listOfIntegralImages);
+		RandomAccessibleInterval<? extends Composite<RealType>> compositeRAI = Views
+			.collapse(stacked);
+		RandomAccessibleInterval<? extends Composite<RealType>> extendedCompositeRAI =
+			removeLeadingZeros(compositeRAI);
+
+		NeighborhoodsIterableInterval<? extends Composite<RealType>> neighborhoods =
+			shape.neighborhoodsSafe(extendedCompositeRAI);
+
+		if (map == null) {
```

Comment:
why can't we do this in the initialize? limitation of framework? which information would you need in the framework? @ctrueden maybe we need the smarter type matching or rather the _fake_ `RAI` etc.